test/e2e/antreapolicy_test.go

```diff
@@ -2559,17 +2559,24 @@ func testFQDNPolicyInClusterService(t *testing.T) {
 		failOnError(waitForResourceReady(service, timeout), t)
 	}
 
-	svcDNSName := func(service *v1.Service) string {
+	svcFQDN := func(service *v1.Service) string {
 		return fmt.Sprintf("%s.%s.svc.cluster.local", service.Name, service.Namespace)
 	}
 
+	svcDNSName := func(service *v1.Service) string {
+		// DNS search list is in the following order:
+		// <namespace>.svc.cluster.local, svc.cluster.local, cluster.local
+		// Ignoring the suffix "svc.cluster.local" when accessing the service can get DNS resolution earlier.
+		return fmt.Sprintf("%s.%s", service.Name, service.Namespace)
+	}
+
 	builder := &ClusterNetworkPolicySpecBuilder{}
 	builder = builder.SetName("test-acnp-fqdn-cluster-svc").
 		SetTier("application").
 		SetPriority(1.0)
 	for idx, service := range services {
-		builder.AddFQDNRule(svcDNSName(service), ProtocolTCP, nil, nil, nil, fmt.Sprintf("r%d", idx*2), []ACNPAppliedToSpec{{NSSelector: map[string]string{"ns": namespaces["y"]}, PodSelector: map[string]string{"pod": "b"}}}, crdv1alpha1.RuleActionReject)
-		builder.AddFQDNRule(svcDNSName(service), ProtocolTCP, nil, nil, nil, fmt.Sprintf("r%d", idx*2+1), []ACNPAppliedToSpec{{NSSelector: map[string]string{"ns": namespaces["z"]}, PodSelector: map[string]string{"pod": "c"}}}, crdv1alpha1.RuleActionDrop)
+		builder.AddFQDNRule(svcFQDN(service), ProtocolTCP, nil, nil, nil, fmt.Sprintf("r%d", idx*2), []ACNPAppliedToSpec{{NSSelector: map[string]string{"ns": namespaces["y"]}, PodSelector: map[string]string{"pod": "b"}}}, crdv1alpha1.RuleActionReject)
```

Comment:
Based on your comment above, should we use `svcDNSName` here?

Reply:
We expect the string to be FQDN, not short names, right? otherwise it should be written with wildcard. The dns go library doesn't append the domains in the search list like what the system resolver does, using short name here would get no resolution result.

Reply:
I see what you mean now. If that's the case why do we need the `svcDNSName` function then?

Reply:
It's used in `podToAddrTestStep` as the server address. As the function comment explains, if we use FQDN as the server address, the client would first try `FQDN.<namespace>.svc.cluster.local`, then `FQDN.svc.cluster.local`, then `FQDN.cluster.local`,  last `FQDN`, likely exceeding the timeout.

Reply:
Got it. Thanks for the explanation. Maybe worth adding to the comment in the `podToAddrTestStep` below that each client Pod's `/etc/resolve.conf` will append `svc.cluster.local` as first dns resolution trial, just for laymans like me to understand this better when we look back at the code.

Reply:
used another way to avoid unnecessary resolution attemps, and add comment to explain it, PTAL